homeassistant/components/light/__init__.py

```diff
@@ -270,24 +273,72 @@ async def async_unload_entry(hass, entry):
     return await hass.data[DOMAIN].async_unload_entry(entry)
 
 
-class Profiles:
-    """Representation of available color profiles."""
+def _coerce_none(value: str) -> None:
+    """Coerce an empty string as None."""
+
+    if not isinstance(value, str):
+        raise vol.Invalid("Expected a string")
+
+    if value:
+        raise vol.Invalid("Not an empty string")
+
+
+@dataclasses.dataclass
+class Profile:
+    """Representation of a profile."""
 
-    SCHEMA = vol.Schema(
+    name: str
+    color_x: Optional[float] = dataclasses.field(repr=False)
+    color_y: Optional[float] = dataclasses.field(repr=False)
+    brightness: Optional[int]
+    transition: Optional[int] = None
+    hs_color: Optional[Tuple[float, float]] = dataclasses.field(init=False)
+
+    _OPT_BRIGHTNESS = vol.Any(cv.byte, _coerce_none)  # pylint: disable=invalid-name
+    _OPT_TRANSITION = vol.Any(  # pylint: disable=invalid-name
+        VALID_TRANSITION, _coerce_none
+    )
+    SCHEMA = vol.Schema(  # pylint: disable=invalid-name
         vol.Any(
-            vol.ExactSequence((str, cv.small_float, cv.small_float, cv.byte)),
+            vol.ExactSequence((str, _coerce_none, _coerce_none, cv.byte)),
```

Comment:
Instead of every permutation of allowed schemas, why not do something like this:

```python
vol.ExactSequence(
  vol.Any(str, _coerce_None),
  vol.Any(cv.small_float, _coerce_None),
  # … etc
)
```

Reply:
Schemas are still some what of black magic to me :)
Let me try your approach, but one of the reasons I've settled on permutations: I couldn't figure out how to make in "inclusive" group of `color_x` and `color_y` params in `vol.ExactSequence` schema. i.e. I wanted either both or none of `color_x` and `color_y` be present in the csv row.  And it also allowed me to have schemas, with at least one mandatory parameter.